.github/workflows/dev_on_pull_request_run_e2e_web.yml

```
@@ -0,0 +1,112 @@
+      - name: Setup java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+      - uses: pass-culture-github-actions/cache@v1.0.0
+        id: yarn-modules-cache
+        with:
+          path: |
+            node_modules
+            ~/.cache/yarn
+          key: v1-yarn-dependency-cache-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            v1-yarn-dependency-cache-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          bucket: ${{ secrets.CACHE_BUCKET_NAME }}
+          workload-identity-provider: ${{ secrets.GCP_EHP_WORKLOAD_IDENTITY_PROVIDER }}
+          service-account: ${{ secrets.GCP_EHP_SERVICE_ACCOUNT }}
+      - name: Yarn install
+        run: yarn install
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Set Maestro Version
+        run: echo "MAESTRO_VERSION=1.40.0" >> $GITHUB_ENV
+
+      - name: Cache Maestro CLI
+        id: maestro-cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.maestro
+          key: ${{ runner.os }}-maestro-${{ env.MAESTRO_VERSION }}
+          restore-keys: |
+            ${{ runner.os }}-maestro-
+
+      - name: Install Maestro CLI
+        if: steps.maestro-cache.outputs.cache-hit!= 'true'
+        run: |
+          echo "Installing Maestro CLI version ${MAESTRO_VERSION}..."
+          curl -Ls "https://get.maestro.mobile.dev" | bash
+          echo "$HOME/.maestro/bin" >> $GITHUB_PATH
+        env:
+          MAESTRO_VERSION: ${{ env.MAESTRO_VERSION }}
+
+      - name: Add Maestro to PATH (from cache)
+        if: steps.maestro-cache.outputs.cache-hit == 'true'
+        run: |
+          echo "Using cached Maestro CLI."
+          echo "$HOME/.maestro/bin" >> $GITHUB_PATH
+          maestro -v # Verify version
+
+      - name: Start Vite server
+        run: nohup yarn start:web:staging --host 0.0.0.0 > vite.log 2>&1 &
+      
+      - name: Debug Vite Response
+        run: |
+          sleep 2
+          curl -v http://localhost:5173/ || true
+          
+      - name: Wait for Vite Server
+        run: |
+          echo "Waiting for server at http://localhost:5173/ ..."
+          timeout=60
+          interval=2
+          elapsed=0
+          success=false
+          while [ $elapsed -lt $timeout ]; do
+            status_code=$(curl -s -o /dev/null -w "%{http_code}" -L http://localhost:5173/)
+            if [ "$status_code" -eq 200 ]; then
+              echo "Server is up! (Status: $status_code)"
+              success=true
+              break
+            else
+              echo "Server not ready yet (Status: $status_code). Retrying in $interval seconds..."
+            fi
+            sleep $interval
+            elapsed=$((elapsed + interval))
+          done
+          if [ "$success" = false ]; then
+            echo "Timed out waiting for server after $timeout seconds."
+            exit 1
+          fi
```

je pense qu'on pourrait remplacer tout ça par une commande curl

je remplace `run: |` par `run: >` https://yaml-multiline.info

```suggestion
        run: >
          echo "Waiting for server at http://localhost:5173/ ..." ;
          curl
            --silent
            --output /dev/null
            --location
            --retry 30
            --retry-max-time 60
            --retry-delay 2
            http://localhost:5173/
```

(j'ai fait à l'arrache, à tester)